File: blook_app_flutter/lib/models/report.dart
class Report {
  Report({
    required this.id,
    required this.userId,
    required this.bookCommentId,
    required this.reportComment,
    required this.typeReport,
    required this.createdDate,
  });
  late final String id;
  late final String userId;
  late final String bookCommentId;
  late final String reportComment;
  late final String typeReport;
  late final String createdDate;
  
  Report.fromJson(Map<String, dynamic> json){
    id = json['id'];
    userId = json['user_id'];
    bookCommentId = json['book_comment_id'];
    reportComment = json['report_comment'];
    typeReport = json['type_report'];
    createdDate = json['created_date'];
  }

  Map<String, dynamic> toJson() {
    final _data = <String, dynamic>{};
    _data['id'] = id;
    _data['user_id'] = userId;
    _data['book_comment_id'] = bookCommentId;
    _data['report_comment'] = reportComment;
    _data['type_report'] = typeReport;
    _data['created_date'] = createdDate;
    return _data;
  }
}

File: blook/src/main/java/com/salesianostriana/blook/models/Report.java
package com.salesianostriana.blook.models;

import com.salesianostriana.blook.enums.Estado;
import com.salesianostriana.blook.enums.TypeReport;
import lombok.*;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.Parameter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.Entity;
import javax.persistence.EntityListeners;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter @Setter
@Builder
@EntityListeners(AuditingEntityListener.class)
public class Report {

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
            name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator",
            parameters = {
                    @Parameter(
                            name = "uuid_gen_strategy_class",
                            value = "org.hibernate.id.uuid.CustomVersionOneStrategy"
                    )
            }
    )
    private UUID id;
    private UUID user_id;
    private UUID book_comment_id;
    private String report_comment;
    private TypeReport type_report;
    @CreatedDate
    private LocalDate created_date;
    @Builder.Default
    private Estado estado = Estado.ABIERTO;
}
